generators/heroku/templates/application-heroku.yml.ejs

```diff
@@ -49,12 +49,8 @@ spring:
         hikari:
             maximumPoolSize: 8
 <%_ } else if (prodDatabaseType === 'mongodb') { _%>
-    data:
-        mongodb:
-            uri: ${MONGODB_URI}?authMode=scram-sha1
+    data.mongodb.uri: ${MONGODB_URI}?authMode=scram-sha1
 <%_ } _%>
 <%_ if (searchEngine === 'elasticsearch') { _%>
-    data:
-        jest:
-            uri: ${BONSAI_URL}
+    data.jest.uri: ${BONSAI_URL}
```

Comment:
Why collapse? I like it, but we don't do this often.

Reply:
collapsing makes the generator code cleaner (we don't have to repeat an `if` check). But maybe we should favor the output looking good. I'm split, but happy to change it.

Reply:
I'm OK with it, as long as it works! :)

Reply:
let's ship this, and i'll revisit the next time i need to tweak this file